motr/m0crate/crate_index.c

```diff
@@ -527,14 +654,14 @@ static struct m0_bufvec* idx_bufvec_alloc(int nr)
 static size_t cr_idx_w_get_value_size(struct cr_idx_w *w)
 {
 	int vlen;
-
-	if (w->wit->record_size < 0)
-		vlen = sizeof(struct m0_fid) +
-			cr_rand_pos_range(w->wit->max_record_size);
+	/* If value_size is random, generate it using rand() */
+	if (w->wit->value_size < 0)
+		vlen = w->wit->key_size +
+			cr_rand_pos_range(w->wit->max_value_size);
```

Comment:
replace
cr_rand_pos_range(w->wit->max_value_size)
with
cr_rand_pos_range(w->wit->max_value_size - w->wit->key_size)
?

Otherwise the result may exceed the upper limit value of w->wit->max_value_size?

Reply:
Fixed.